elements/includes/creative_commons.inc

```diff
@@ -72,269 +124,1089 @@ function xml_form_elements_creative_commons($element, &$form_state) {
+    if (!in_array($jurisdiction, $supported_jurisdictions) || $jurisdiction == $supported_jurisdictions[0]) {
       $jurisdiction = '';
     }
     else {
-      // If not international add the trailing slash.
       $jurisdiction .= '/';
     }
-    return "http://creativecommons.org/licenses/by$arguments/$license_version/$jurisdiction";
   }
+  // Get the specified version.
+  $version = $values['license_fieldset']['license_version'];
+  // If no version was specified, default to versions that should be available.
+  if (empty($version)) {
+    $version = ($license == 'by') ? ($jurisdiction == '' ? '4.0' : '2.5') : '1.0';
+  }
+  // Return the license URI.
+  return XML_FORM_ELEMENTS_CREATIVE_COMMONS_BASE_URL . "$path/$license$arguments/$version/$jurisdiction";
+}
+
+/**
+ * Get markup and values for `creative_commons` form element.
+ *
+ * @param array $element
+ *   The `creative_commons` form element.
+ * @param array $form_state
+ *   The form state.
+ *
+ * @return array
+ *   An array containing:
+ *   - the markup for the chosen license;
+ *   - the associative value array for that license.
+ */
+function xml_form_elements_creative_commons_get_processed_values(&$element, &$form_state) {
+  // Get current form values or defaults.
+  $values = isset($form_state['input'][$element['#name']]) ?
+    xml_form_elements_creative_commons_cleanup_input(is_string($form_state['values'][$element['#name']]) ?
+      xml_form_elements_creative_commons_get_values_from_uri($form_state['values'][$element['#name']]) :
+      $form_state['values'][$element['#name']]) :
+    xml_form_elements_creative_commons_get_values_from_uri(isset($element['#default_value']) ?
+      $element['#default_value'] :
+      '');
+
+  // Set prefix for license HTML markup.
+  $cc_prefix = '<strong>' . t('Selected license:') . '</strong> ';
+
+  // Define none markup.
+  $none = t('None.');
+
+  // Create empty values.
+  $empty_values = xml_form_elements_creative_commons_get_values_from_uri('');
+
+  // If some values are missing, return filled-up value array.
+  $missing_values = array('license_fieldset' => array_diff_key($empty_values['license_fieldset'], $values['license_fieldset']));
+  if (!isset($values['license_fieldset']) || !empty($missing_values['license_fieldset'])) {
+    $values['license_fieldset'] = array_merge($empty_values['license_fieldset'], isset($values['license_fieldset']) ? $values['license_fieldset'] : array());
+    drupal_set_message(t('An internal error occurred while processing the Creative Commons form element "@display_name" (unexpected data loss). Please inform the site administrator.', array("@display_name" => $element['license_fieldset']['#title'])), 'error');
+    watchdog('xml_form_elements', 'Unexpected data loss when processing element "@element_name" in "@form_name". Probably you just found a bug.', array('@element_name' => $element['#name'], '@form_name' => $form_state['storage'][FormStorage::STORAGE_ROOT]['name']), WATCHDOG_ERROR);
```

Comment:
Just to be sure @d-r-p, did you test that `FormStorage`class is available in this scope?

Reply:
@DiegoPino well, I did nuke an array element to trigger the error for testing and it seemed to work just fine...

Reply:
Can you confirm this is also the case in the form builder UI  as it is on a live XML form multi step ingest? Sorry for being overprotective...

Reply:
@DiegoPino I think I _did_ test this in both scenarios (form builder view form + ingest new object), but will re-test to be extra sure.

Reply:
@DiegoPino I tested scope availability by triggering watchdog in [L347](https://github.com/Islandora/islandora_xml_forms/pull/253/files/f6a0d63b322aa7613b20d266d219090c7cdbb4ef#diff-0fca5b132da440f97135da3015937c40R347). Both scenarios seem to have `FormStorage` available. Should the other two occurrences be tested separately, or are we good?